internal/interfaces/http/tdex_connect_handler.go

```diff
@@ -46,52 +54,97 @@ func NewTdexConnectService(
 		serverPort:        serverPort,
 		macaroonPath:      macaroonPath,
 		certPath:          certPath,
+		repoManager:       repoManager,
 	}
 }
 
 func (t *tdexConnect) TdexConnectHandler(w http.ResponseWriter, req *http.Request) {
-	styleFile, err := ioutil.ReadFile(tdexConnectTemplateCssFile)
-	if err != nil {
-		log.Errorln(err.Error())
-
-		http.Error(w, http.StatusText(500), 500)
-		return
-	}
-
-	macaroon := t.macaroonBytes
-	if t.macaroonBytes == nil {
-		macaroon = readFile(t.macaroonPath)
-		t.macaroonBytes = macaroon
-	}
-
+	ctx := context.Background()
+	walletStatus := t.walletUnlockerSvc.IsReady(ctx)
 	cert := t.certBytes
 	if t.certBytes == nil {
 		cert = readFile(t.certPath)
 		t.certBytes = cert
 	}
 
 	connectUrl, err := tdexdconnect.EncodeToString(
-		t.serverPort, cert, macaroon,
+		t.serverPort, cert, nil,
 	)
 	if err != nil {
 		log.Errorln(err.Error())
 		http.Error(w, http.StatusText(500), 500)
 	}
 
-	data := Page{
-		Title: tdexConnectHtmlTitle,
-		Style: template.CSS(styleFile),
-		Url:   template.URL(connectUrl),
-	}
-
-	//walletStatus := t.walletUnlockerSvc.IsReady(context.Background())
-	//if walletStatus.Initialized {
-	//	TODO check password
-	//}
+	method := req.Method
+	if method == http.MethodGet {
```

Comment:
This way we must pass the password in a HTTP call like POST?

Reply:
Yes password is passed using POST, check [js](https://github.com/sekulicd/tdex-daemon/blob/tdexconnect_url_html/web/app.js) file i added.

Reply:
I would rather branch here on the `walletStatus.Initialized` instead

if initialized we then require a basic authentication header with the password, returning status code 401 if password wrong or not presente. If password is given and correct, we render the HTML page with the complete URL

If not initialized, we simply return the URL, no need for authentication.